server/uploading.go

```diff
@@ -42,18 +42,17 @@ type uploadTracker struct {
 	uploading     map[string]*put.Request
 	stuckRequests map[string]*put.Request
 
-	slacker  set.Slacker
-	debounce time.Duration
-	bouncing bool
-	lastMsg  string
+	slackConfig SlackConfig
 }
 
 func newUploadTracker(slacker set.Slacker, debounce time.Duration) *uploadTracker {
 	ut := &uploadTracker{
 		uploading:     make(map[string]*put.Request),
 		stuckRequests: make(map[string]*put.Request),
-		slacker:       slacker,
-		debounce:      debounce,
+		slackConfig: SlackConfig{
```

Comment:
This is backwards. A new() method would take a config struct to make it easy for users to supply many options, some of which were optional, and easy to add extra options in the future without changing the signature of the new method.

It would then store desired things from the config directly on the returned struct's properties, as we had here originally, so that they are private and can't change out from under us during run time if the user alters their config struct. We only might store the config struct if it had complex methods we needed to call later and can't call now inside the new().